tests/integration/test_database.py

```diff
@@ -82,6 +90,131 @@ async def test_build_and_deploy(ops_test: OpsTest):
 @pytest.mark.order(2)
 @pytest.mark.abort_on_fail
 @pytest.mark.database_tests
+async def test_password_rotation(ops_test: OpsTest):
+    """Rotate password and confirm changes."""
+    num_units = len(ops_test.model.applications[DATABASE_APP_NAME].units)
+    random_unit = ops_test.model.applications[DATABASE_APP_NAME].units[num_units - 1]
+
+    old_credentials = await fetch_credentials(random_unit, SERVER_CONFIG_USERNAME)
+
+    primary_unit = await get_primary_unit(
+        ops_test,
+        random_unit,
+        DATABASE_APP_NAME,
+        CLUSTER_NAME,
+        old_credentials["username"],
+        old_credentials["password"],
+    )
+    primary_unit_address = await primary_unit.get_public_address()
+    logger.debug(
+        "Test succeeded Primary unit detected before password rotation is %s", primary_unit_address
+    )
+
+    new_password = generate_random_password(PASSWORD_LENGTH)
+
+    await rotate_credentials(
+        unit=primary_unit, username=SERVER_CONFIG_USERNAME, password=new_password
+    )
+
+    updated_credentials = await fetch_credentials(random_unit, SERVER_CONFIG_USERNAME)
+    assert updated_credentials["password"] != old_credentials["password"]
+    assert updated_credentials["password"] == new_password
+
+    primary_unit = await get_primary_unit(
+        ops_test,
+        random_unit,
+        DATABASE_APP_NAME,
+        CLUSTER_NAME,
+        updated_credentials["username"],
+        updated_credentials["password"],
+    )
+    primary_unit_address = await primary_unit.get_public_address()
+    logger.debug(
+        "Test succeeded Primary unit detected after password rotation is %s", primary_unit_address
+    )
+
+
+@pytest.mark.order(3)
+@pytest.mark.abort_on_fail
+@pytest.mark.database_tests
+async def test_password_rotation_silent(ops_test: OpsTest):
+    """Rotate password and confirm changes."""
+    num_units = len(ops_test.model.applications[DATABASE_APP_NAME].units)
+    random_unit = ops_test.model.applications[DATABASE_APP_NAME].units[num_units - 1]
+
+    old_credentials = await fetch_credentials(random_unit, SERVER_CONFIG_USERNAME)
+
+    primary_unit = await get_primary_unit(
+        ops_test,
+        random_unit,
+        DATABASE_APP_NAME,
+        CLUSTER_NAME,
+        old_credentials["username"],
+        old_credentials["password"],
+    )
+    primary_unit_address = await primary_unit.get_public_address()
+    logger.debug(
+        "Test succeeded Primary unit detected before password rotation is %s", primary_unit_address
+    )
+
+    await rotate_credentials(unit=primary_unit, username=SERVER_CONFIG_USERNAME)
+
+    updated_credentials = await fetch_credentials(random_unit, SERVER_CONFIG_USERNAME)
+    assert updated_credentials["password"] != old_credentials["password"]
+
+    primary_unit = await get_primary_unit(
```

Comment:
same thing as above, do we need lines 165-175 in this test?

Reply:
ack